contracts/strategies/ltip-simple/LTIPSimpleStrategy.sol

```diff
@@ -0,0 +1,708 @@
+        internal
+        virtual
+    {
+        TokenTimelock vestingContract =
+            new TokenTimelock(IERC20(_token), recipientAddress, block.timestamp + vestingPeriod);
+
+        _transferAmount(_token, address(vestingContract), _amount);
+
+        _vestingPlans[recipientId] = VestingPlan(address(vestingContract), 0);
+
+        emit VestingPlanCreated(recipientId, address(vestingContract), 0);
+    }
+
+    /// @notice Distribute the upcoming milestone to acceptedRecipientId.
+    /// @dev As allocation determines the acceptance, anybody should be able to distribute.
+    /// @param _sender The sender of the distribution.
+    function _distribute(address[] memory __recipientIds, bytes memory, address _sender) internal virtual override {
+        IAllo.Pool memory pool = allo.getPool(poolId);
+
+        uint256 recipientLength = __recipientIds.length;
+
+        for (uint256 i; i < recipientLength;) {
+            address recipientId = __recipientIds[i];
+
+            Recipient memory recipient = _recipients[recipientId];
+
+            // Check if the recipient is accepten
+            if (recipient.recipientStatus != Status.Accepted) revert RECIPIENT_NOT_ACCEPTED();
+            if (votes[recipientId] < votingThreshold) revert INSUFFICIENT_VOTES();
+
+            if (_vestingPlans[recipientId].vestingContract != address(0)) revert ALREADY_ALLOCATED();
+
+            // Get the pool, subtract the amount and transfer to the recipient
+            poolAmount -= recipient.allocationAmount;
+
+            _vestAmount(recipientId, recipient.recipientAddress, pool.token, recipient.allocationAmount);
+
+            // Emit events for the milestone and the distribution
+            emit Distributed(recipientId, recipient.recipientAddress, recipient.allocationAmount, _sender);
+
+            unchecked {
+                ++i;
+            }
+        }
+    }
+
+    function _isPoolTimestampValid(
+        uint64 _allocationStartTime,
+        uint64 _allocationEndTime,
+        uint64 _distributionStartTime,
+        uint64 _distributionEndTime,
+        uint64 _registrationStartTime,
+        uint64 _registrationEndTime,
+        uint64 _reviewStartTime,
+        uint64 _reviewEndTime
+    ) internal view {
+        if (
+            block
+                // Register timestamps must be in the future
+                .timestamp > _registrationStartTime
+            // Start times must be before end times
+            || _allocationStartTime > _allocationEndTime || _distributionStartTime > _distributionEndTime
+                || _registrationStartTime > _registrationEndTime || _reviewStartTime > _reviewEndTime
+            // Some end times must be after other end times
+            || _registrationEndTime > _allocationEndTime || _allocationEndTime > _distributionStartTime
+                || _reviewStartTime > allocationEndTime
+        ) {
+            revert INVALID();
+        }
```

Comment:
This `if` is huge! It is also hard to pinpoint the issue with the specific error message and the different branches of the if. I think it makes more sense to create multiple single `if`s for each start-end time with their custom errors. This way it will make it much easier to test aswell